daac-ops-api/lambdas/sf-sns-broadcast.js

```diff
@@ -35,8 +35,8 @@ async function publish(message, finish = false) {
   }
 
   if (failed) {
-    const error = get(event, 'exception.Error');
-    const cause = get(event, 'exception.Cause');
+    const error = get(event, 'exception.Error', get(event, 'error.Error'));
```

Comment:
Could we update this to handle all of these cases?

I wonder if there is a fix for the actual problem which is Step Function puts the error object in the exception in its entirety.

```yaml
ModisProcess:
        Type: Task
        Resource: ${ModisProcessingLambdaFunction.Arn}
        Catch:
          - ErrorEquals:
            - States.ALL
            ResultPath: '$.exception'
            Next: StopStatus
        Retry:
          - ErrorEquals:
              - States.ALL
            IntervalSeconds: 2
            MaxAttempts: 3
        Next: CmrStep
```

Reply:
So with these changes it can handle both `ResultPath: '$.error'` and `ResultPath: '$.exception'`. Do you think if we set `ResultPath: '$'` it will end up at the top-level object in respective Error and Cause fields?

Reply:
We have to test it. Although with the new message adapter, this is no longer going to be needed.